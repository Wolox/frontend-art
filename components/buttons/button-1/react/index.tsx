import React from 'react';

import '../styles.scss';

function Button1() {
  return (
    <div className="column center btn-group-1">
      <button className="btn btn-1 m-bottom-2" disabled={true}>
        <div className="btn__content">cancel</div>
      </button>
      <button type="button" className="btn btn-1">
        <div className="btn__content">submit</div>
      </button>
    </div>
  );
}

export default Button1;
