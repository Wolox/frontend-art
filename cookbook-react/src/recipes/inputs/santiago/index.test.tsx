import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import Example from './index';

test('renders the example correctly', () => {
  const { container } = render(<Example />);
  expect(container).toMatchSnapshot();
});

test('can click example 1', () => {
  render(<Example />);
  const tab = screen.getAllByText('Opción 3')[0];
  userEvent.click(tab);
  expect(tab).toHaveClass('active');
});

test('can click example 2', () => {
  render(<Example />);
  const tab = screen.getAllByText('Opción 3')[1];
  userEvent.click(tab);
  expect(tab).toHaveClass('active');
});

test('can click external button in example 2', () => {
  render(<Example />);
  const option1Tab = screen.getAllByText('Opción 1')[1];
  const option2Tab = screen.getAllByText('Opción 2')[1];
  const option3Tab = screen.getAllByText('Opción 3')[1];
  const button = screen.getByText('Go to next tab');

  expect(option1Tab).not.toHaveClass('active');
  expect(option2Tab).toHaveClass('active');
  expect(option3Tab).not.toHaveClass('active');

  userEvent.click(button);

  expect(option1Tab).not.toHaveClass('active');
  expect(option2Tab).not.toHaveClass('active');
  expect(option3Tab).toHaveClass('active');

  userEvent.click(button);

  expect(option1Tab).toHaveClass('active');
  expect(option2Tab).not.toHaveClass('active');
  expect(option3Tab).not.toHaveClass('active');
});

test('can click example 3', () => {
  render(<Example />);
  const tab = screen.getAllByText('Opción 2')[2];
  userEvent.click(tab);
  expect(tab).toHaveClass('active');
});
