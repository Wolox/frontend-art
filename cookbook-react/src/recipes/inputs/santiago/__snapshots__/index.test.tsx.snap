// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the example correctly 1`] = `
<div>
  <div
    class="exampleContainer"
  >
    <h2
      class="title"
    >
      Active state controlled by the Selector component
    </h2>
    <div
      class="optionsContainer tab"
    >
      <label
        class="optionLabel"
        for="option-1"
      >
        <input
          class="option"
          id="option-1"
          name="option"
          type="radio"
          value="1"
        />
        Opción 1
      </label>
      <div
        class="separator active"
      />
      <label
        class="optionLabel active"
        for="option-2"
      >
        <input
          checked=""
          class="option"
          id="option-2"
          name="option"
          type="radio"
          value="2"
        />
        Opción 2
      </label>
      <div
        class="separator"
      />
      <label
        class="optionLabel"
        for="option-3"
      >
        <input
          class="option"
          id="option-3"
          name="option"
          type="radio"
          value="3"
        />
        Opción 3
      </label>
    </div>
    <h3
      class="firstContent"
    >
      Selected tab with id: 1
    </h3>
    <h2
      class="title"
    >
      Active state controlled by the Selector's parent component
    </h2>
    <div
      class="optionsContainer tab"
    >
      <label
        class="optionLabel"
        for="option-10"
      >
        <input
          class="option"
          id="option-10"
          name="option"
          type="radio"
          value="10"
        />
        Opción 1
      </label>
      <div
        class="separator active"
      />
      <label
        class="optionLabel active"
        for="option-20"
      >
        <input
          checked=""
          class="option"
          id="option-20"
          name="option"
          type="radio"
          value="20"
        />
        Opción 2
      </label>
      <div
        class="separator"
      />
      <label
        class="optionLabel"
        for="option-30"
      >
        <input
          class="option"
          id="option-30"
          name="option"
          type="radio"
          value="30"
        />
        Opción 3
      </label>
    </div>
    <h3>
      Selected tab with id: 20
    </h3>
    <button
      class="button"
      type="button"
    >
      Go to next tab
    </button>
    <h2
      class="title"
    >
      Selector with disabled tab
    </h2>
    <div
      class="optionsContainer tab"
    >
      <label
        class="optionLabel"
        for="option-id1"
      >
        <input
          class="option"
          id="option-id1"
          name="option"
          type="radio"
          value="id1"
        />
        Opción 1
      </label>
      <div
        class="separator"
      />
      <label
        class="optionLabel"
        for="option-id2"
      >
        <input
          class="option"
          id="option-id2"
          name="option"
          type="radio"
          value="id2"
        />
        Opción 2
      </label>
      <div
        class="separator"
      />
      <label
        class="optionLabel disabled"
        for="option-id3"
      >
        <input
          class="option"
          disabled=""
          id="option-id3"
          name="option"
          type="radio"
          value="id3"
        />
        Opción 3
      </label>
    </div>
    <h3
      class="firstContent"
    >
      Selected tab with id: id1
    </h3>
  </div>
</div>
`;
